### PR TITLE
Add OpenComputers support

### PR DESCRIPTION
**Pull request by MineRobber9000**
This PR adds support for OpenComputers. It's ~~almost~~ complete. ~~All I need to implement for complete compatability:~~

 * [X] ~~`fs.currentProgram`~~
 * [X] ~~`http.request`~~

~~The latter will be easy, and the former should be easy with more research. I figured I'd get my foot in the door as I finish up.~~

It's now complete and ready for review! @SquidDev can you review this please?

### CODE REVIEW

File: howl/platform/oc.lua [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  local size = fh:seek("end")
+  fh:close()
+  return size
+end
+
+local function request(url,post,headers)
+  if not hasInternet then error("No internet card found",0) end
+	local resp = ""
+  for chunk in internet.request(url,post,headers) do
+    resp += chunk
```
Comment (SquidDev) [inline howl/platform/oc.lua]: This is definitely not valid Lua. I'm presuming it should be `resp = resp .. chunk`?

File: howl/platform/oc.lua [hunk 2]
```diff
@@ -0,0 +1,129 @@
+--- OpenComputers's platform table
```
Comment (SquidDev) [inline howl/platform/oc.lua]: Would it be possible to indent this with tabs? I know Lua is "meant" to be indented with 2 spaces, but it's consistent with the rest of the project.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Thanks for the PR. I'm so sorry for taking this long to get to this. I know it's really frustrating seeing one's work ignored.

I've encountered a couple of issues with testing this, which I've commented on. If you're still interested in this PR, would it be possible to fix them? If not (and I can hardly blame you) I'll fix them myself.

File: howl/platform/oc.lua (hunk 1 above)
Comment (MineRobber9000) [reply howl/platform/oc.lua]: yeah, I messed up. I'll get around to fixing it soon.

File: howl/platform/oc.lua (hunk 2 above)
Comment (MineRobber9000) [reply howl/platform/oc.lua]: I'll go through and indent with tabs then. I just wasn't sure what you wanted for indentation.

**Comment by MineRobber9000** [comment]
While yes, it has been a while, (almost 3 months) I am still interested in making this work.

**Comment by SquidDev** [comment]
And merged. Thank you very much for all your work. Once again, I'm so sorry for the delay.

**Comment by MineRobber9000** [comment]
It's okay. Thanks for fixing the else/if. (even if I would've fixed it if you included it in your review :P)

**Pull request merged**